src/Components/Artwork/Details/__tests__/SaleMessage.jest.tsx

```diff
@@ -0,0 +1,235 @@
+import { DateTime } from "luxon"
+import { Details_artwork$data } from "__generated__/Details_artwork.graphql"
+import { SaleMessageQueryRenderer } from "Components/Artwork/Details/SaleMessage"
+import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils"
+import { createRef } from "react"
+import { act, render, screen } from "@testing-library/react"
+import { useDidMount } from "Utils/Hooks/useDidMount"
+import { useIntersectionObserver } from "Utils/Hooks/useIntersectionObserver"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+import { CleanRelayFragment } from "Utils/typeSupport"
+
+jest.unmock("react-relay")
+
+jest.mock("System/Hooks/useSystemContext")
+jest.mock("Utils/Hooks/useDidMount")
+jest.mock("Utils/Hooks/useIntersectionObserver")
+
+describe("SaleMessage", () => {
+  let mockuseDidMount = useDidMount as jest.Mock
+  let mockUseIntersectionObserver = useIntersectionObserver as jest.Mock
+  let mockUseSystemContext = useSystemContext as jest.Mock
+  let mockEnvironment
+
+  beforeEach(() => {
+    mockuseDidMount.mockImplementation(() => true)
+
+    let called = false
+    mockUseIntersectionObserver.mockImplementation(({ onIntersection }) => {
+      if (!called) {
+        onIntersection()
+        called = true
+      }
+      return { ref: createRef() }
+    })
+
+    mockEnvironment = createMockEnvironment()
+    mockUseSystemContext.mockImplementation(() => {
+      return {
+        relayEnvironment: mockEnvironment,
+      }
+    })
+  })
+
+  describe("artwork in active auction", () => {
+    describe("after fetching null partner offer", () => {
+      const renderSaleMessage = artwork => {
+        render(
+          <SaleMessageQueryRenderer
+            artwork={artwork}
+            id={"opaque-internal-id"}
+          />
+        )
+
+        act(() => {
+          mockEnvironment.mock.resolveMostRecentOperation(operation =>
```

Comment:
This allows us to skip manually mocking both the test environment and the relay operation; all of those things happen inside of our test tools automatically.